src/GraphQL/StringExtensions.cs

```diff
@@ -70,33 +70,47 @@ private static string FastChangeFirstLetter(char newFirstLetter, string s)
         /// </summary>
         public static string ToConstantCase(this string value)
         {
-            //aka: return Regex.Replace(s, @"([A-Z])([A-Z][a-z])|([a-z0-9])([A-Z])", "$1$3_$2$4").ToUpperInvariant();
             int i;
             int strLength = value.Length;
             // iterate through each character in the string, stopping a character short of the end
             for (i = 0; i < strLength - 1; ++i)
             {
                 var curChar = value[i];
                 var nextChar = value[i + 1];
-                // look for the pattern [a-z0-9][A-Z]
-                if ((char.IsLower(curChar) || char.IsDigit(curChar)) && char.IsUpper(nextChar))
+                // look for the pattern [a-z][A-Z]
+                if (char.IsLower(curChar) && char.IsUpper(nextChar))
                 {
-                    // add an underscore between the two characters, increment i to skip the underscore, and increase strLength because the string is longer now
-                    value = value.Substring(0, ++i) + '_' + value.Substring(i);
-                    ++strLength;
-                    // skip the following match check since we already found a match here
+                    InsertUnderscore();
+                    continue;
+                }
+                // look for the pattern [0-9][A-Za-z]
+                if (char.IsDigit(curChar) && char.IsLetter(nextChar))
+                {
+                    InsertUnderscore();
+                    continue;
+                }
+                // look for the pattern [A-Za-z][0-9]
+                if (char.IsLetter(curChar) && char.IsDigit(nextChar))
+                {
+                    InsertUnderscore();
                     continue;
                 }
                 // if there's enough characters left, look for the pattern [A-Z][A-Z][a-z]
                 if (i < strLength - 2 && char.IsUpper(curChar) && char.IsUpper(nextChar) && char.IsLower(value[i + 2]))
                 {
-                    // add an underscore between the two characters, increment i to skip the underscore, and increase strLength because the string is longer now
-                    value = value.Substring(0, ++i) + '_' + value.Substring(i);
-                    ++strLength;
+                    InsertUnderscore();
```

Comment:
```suggestion
                    InsertUnderscore();
                    continue;
```

to unify all cases ?